src/cluster.c

```diff
@@ -4948,6 +4962,163 @@ void addNodeReplyForClusterSlot(client *c, clusterNode *node, int start_slot, in
     setDeferredArrayLen(c, nested_replylen, nested_elements);
 }
 
+void addNodeDetailsToShardReply(client *c, clusterNode *node) {
+    /*
+     * Following information regarding a node is returned.
+     *
+     * 1. id (40 char len)
+     * 2. port
+     * 3. ip
+     * 4. hostname
+     * 5. endpoint
+     * 5. role
+     * 6. replication offset (in case of replica only).
+     * 7. health
+     */
```

Comment:
I think this documentation is all that helpful, since it's pretty easy to scan through and see this all.

Reply:
I used the above while writing the below code 😄